client/ayon_substancepainter/plugins/publish/collect_staging_dir_for_textures.py

```diff
@@ -0,0 +1,39 @@
+import copy
+
+import pyblish.api
+from ayon_core.pipeline import publish
+
+
+class CollectStagingDirExportConfig(pyblish.api.InstancePlugin):
+    """Collect Staging Dir for Export Config to export the textures"""
+
+
+    label = "Collect Staging dir for Export Config"
+    hosts = ["substancepainter"]
+    families = ["textureSet", "textures", "image"]
+    order = pyblish.api.CollectorOrder + 0.491
+
+    def process(self, instance):
+        export_config = copy.deepcopy(instance.data["exportConfig"])
+        export_config["exportPath"] = publish.get_instance_staging_dir(instance)
+        instance.data["exportConfig"] = export_config
+        instance.data.update(instance.data["exportConfig"])
+
+class CollectStagingDirTexture(pyblish.api.InstancePlugin):
+    """Collect Staging Dir as the representation data for the texture publish"""
+
+
+    label = "Collect Staging dir for texture"
+    hosts = ["substancepainter"]
+    families = ["image", "textures"]
+    order = pyblish.api.CollectorOrder + 0.4911
+
+    def process(self, instance):
+        representations: "list[dict]" = instance.data["representations"]
+        staging_dir = instance.data["exportConfig"]["exportPath"]
+        updated_representations = []
+        for representation in list(representations):
+            tmp_representation = copy.deepcopy(representation)
+            tmp_representation["stagingDir"] = staging_dir
+            updated_representations.append(tmp_representation)
+        instance.data["representations"] = updated_representations
```

Comment:
No need to make deep copies - in this case we DO want to update the representations in-place. Because we want to fix the existing entries explicitly.

```suggestion
        # Update the collected staging dir because we initially
        # collected the textures sets using a temp directory
        # to allow the instances to be defined prior to defining
        # their expected paths (staging dir) which may be based
        # on anatomy data and custom staging dirs, etc.
        representations: "list[dict]" = instance.data["representations"]
        for repre in instance.data["representations"]:
            repre["stagingDir"] = instance.data["stagingDir"]
```

Also, still feels nasty as hell ;) but we can maybe improve later by not collecting the representations earlier - but only collect them at this order so we don't need to "update" them but can just start collecting them when we have the relevant staging dir data from the instances.